lib/Team.js

```diff
@@ -0,0 +1,172 @@
+    * @return {Promise} - the promise for the http request
+    */
+   getMembership(username, cb) {
+      log(`Getting membership of user ${username} in Team ${this.__teamId}`);
+      return this._request('GET', `/teams/${this.__teamId}/memberships/${username}`, undefined, cb);
+   }
+
+   /**
+    * Add a member to the Team
+    * @see https://developer.github.com/v3/orgs/teams/#add-team-membership
+    * @param {string} username - can be one of: `all`, `maintainer`, or `member`
+    * @param {object} options - Parameters for adding a team member
+    * @param {string} [options.role=member] - The role that this user should have in the team. Can be one
+    * of: `member`, or `maintainer`
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   addUser(username, options, cb) {
+      log(`Adding user ${username} to Team ${this.__teamId}`);
+      return this._request('PUT', `/teams/${this.__teamId}/memberships/${username}`, options, cb);
+   }
+
+   /**
+    * Add a member to the Team
+    * @see https://developer.github.com/v3/orgs/teams/#remove-team-membership
+    * @param {string} username - can be one of: `all`, `maintainer`, or `member`
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   removeUser(username, cb) {
+      log(`Removing user ${username} from Team ${this.__teamId}`);
+      return this._request204or404(`/teams/${this.__teamId}/memberships/${username}`, undefined, cb, 'DELETE');
+   }
+
+   /**
+    * Get repo management status for team
+    * @see https://developer.github.com/v3/orgs/teams/#remove-team-membership
+    * @param {string} owner - Organization name
+    * @param {string} repo - Repo name
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   managesRepo(owner, repo, cb) {
+      log(`Getting repo management by Team ${this.__teamId} for repo ${owner}/${repo}`);
+      return this._request204or404(`/teams/${this.__teamId}/repos/${owner}/${repo}`, undefined, cb);
+   }
+
+   /**
+    * Add or Update repo management status for team
+    * @see https://developer.github.com/v3/orgs/teams/#add-or-update-team-repository
+    * @param {string} owner - Organization name
+    * @param {string} repo - Repo name
+    * @param {object} options - Parameters for adding or updating repo management for the team
+    * @param {string} [options.permission] - The permission to grant the team on this repository. Can be one
+    * of: `pull`, `push`, or `admin`
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   addOrUpdateRepo(owner, repo, options, cb) {
```

Comment:
This one is tough, `update` gives a feeling that the repo is already managed, which may not be the case.